apm-sniffer/apm-sdk-plugin/solrj-7.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/solrj/SolrClientInterceptor.java

```diff
@@ -0,0 +1,190 @@
+
+                        List<String> deleteBy = ur.getDeleteById();
+                        if (deleteBy == null) {
+                            actionName = "DELETE_BY_QUERY";
+                            deleteBy = ur.getDeleteQuery();
+                        }
+                        if (deleteBy == null) {
+                            deleteBy = new ArrayList<String>();
+                        }
+                        span = getSpan(collection, request.getPath(), actionName, instance.getRemotePeer());
+                        span.tag(SolrjTags.TAG_DELETE_VALUE, deleteBy.toString());
+                    } else {
+                        span = getSpan(collection, request.getPath(), "ADD", instance.getRemotePeer());
+                        span.tag(SolrjTags.TAG_DOCS_SIZE, String.valueOf(documents.size()));
+                        span.tag(SolrjTags.TAG_COMMIT_WITHIN, String.valueOf(update.getCommitWithin()));
+                    }
+                } else {
+                    span = getSpan(collection, request.getPath(), instance.getRemotePeer());
+                }
+            } else {
+                if (action == AbstractUpdateRequest.ACTION.COMMIT) {
+                    span = getSpan(collection, request.getPath(), action.name(), instance.getRemotePeer());
+                    span.tag(SolrjTags.TAG_SOFT_COMMIT, params.get(UpdateParams.SOFT_COMMIT, ""));
+                } else {
+                    span = getSpan(collection, request.getPath(), action.name(), instance.getRemotePeer());
+                    span.tag(SolrjTags.TAG_MAX_OPTIMIZE_SEGMENTS, params.get(UpdateParams.MAX_OPTIMIZE_SEGMENTS, "1"));
+                }
+            }
+        } else if (request instanceof QueryRequest) {
+            span = getSpan(collection, request.getPath(), instance.getRemotePeer());
+
+            span.tag(SolrjTags.TAG_SORT_BY, params.get(CommonParams.SORT, ""));
+            span.tag(SolrjTags.TAG_START, params.get(CommonParams.START, "0"));
+            span.tag(SolrjTags.TAG_QT, params.get(CommonParams.QT, request.getPath()));
+        } else {
+            span = getSpan(collection, request.getPath(), instance.getRemotePeer());
+        }
+        span.tag(SolrjTags.TAG_COLLECTION, collection);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
+                              Class<?>[] argumentsTypes, Object ret) throws Throwable {
+        if (!ContextManager.isActive()) {
+            return ret;
+        }
+
+        AbstractSpan span = ContextManager.activeSpan();
+        NamedList<Object> result = (NamedList<Object>) ret;
+        if (result != null) {
+            NamedList<Object> header = (NamedList<Object>) result.get("responseHeader");
+            if (header != null) {
+                span.tag(SolrjTags.TAG_STATUS, String.valueOf(header.get("status")));
```

Comment:
By following your explanation, this tag value should work with `span#errorOccurred` together.

Reply:
It is ok.